core/types/transaction_marshalling.go

```diff
@@ -223,10 +223,10 @@ func (t *Transaction) UnmarshalJSON(input []byte) error {
 			return errors.New("missing required field 'etxIndex' in external transaction")
 		}
 		etx.ETXIndex = uint16(*dec.ETXIndex)
-		if dec.Gas == nil {
-			return errors.New("missing required field 'gas' in external transaction")
+		if dec.Gas != nil {
+			//return errors.New("missing required field 'gas' in external transaction")
+			etx.Gas = uint64(*dec.Gas)
```

Comment:
What is the purpose of this PR? AFAICT the "missing required field" error was correct, but perhaps I am missing something?